storage/src/adb/immutable/mod.rs

```diff
@@ -321,7 +319,12 @@ impl<E: RStorage + Clock + Metrics, K: Array, V: Codec, H: CHasher, T: Translato
                             log_size = section * log_items_per_section.get();
                             oldest_retained_loc = Some(log_size);
                         }
+
                         let loc = log_size; // location of the current operation.
+                        if after_last_commit.is_none() {
+                            after_last_commit = Some((loc, offset));
```

Comment:
I suppose this could be `offset=0`? I think we should keep the blob around in that case but just raising if we should think more about that.